docker/scripts/entrypoint.sh

```diff
@@ -20,92 +30,51 @@ NH_SECRET_PHRASE_PATH=${NH_SECRET_PHRASE_PATH:-"/data/secret_phrase.dat"}
 echo "NH_SECRET_PHRASE_PATH=${NH_SECRET_PHRASE_PATH}"
 
 # Node configurations (env->arg)
-NH_CONF_NAME=${NH_CONF_NAME:-"MyNode"}
-NH_CONF_BASE_PATH=${NH_CONF_BASE_PATH:-"/data/node"}
-NH_CONF_CHAIN=${NH_CONF_CHAIN:-"test"}
-NH_CONF_VALIDATOR=${NH_CONF_VALIDATOR:-}
-NH_CONF_NODE_KEY_FILE=${NH_CONF_NODE_KEY_FILE:-}
-NH_CONF_BOOTNODES=${NH_CONF_BOOTNODES:-}
-NH_CONF_RPC_CORS=${NH_CONF_RPC_CORS:-}
-NH_CONF_RPC_EXTERNAL=${NH_CONF_RPC_EXTERNAL:-}
-NH_CONF_RPC_METHODS=${NH_CONF_RPC_METHODS:-}
-NH_CONF_PRUNING=${NH_CONF_PRUNING:-}
-NH_PROMETHEUS_EXTERNAL=${NH_PROMETHEUS_EXTERNAL:-}
-
-for var_name in NH_CONF_NAME NH_CONF_BASE_PATH NH_CONF_CHAIN NH_CONF_VALIDATOR NH_CONF_NODE_KEY_FILE NH_CONF_BOOTNODES NH_CONF_RPC_CORS NH_CONF_RPC_EXTERNAL NH_CONF_RPC_METHODS NH_CONF_PRUNING NH_PROMETHEUS_EXTERNAL; do
-  # Get the value of the variable
-  var_value="${!var_name}"
-
-  # Check if the variable is defined
-  if [ -n "${var_value}" ]; then
-    echo "${var_name}=${var_value}"
-  else
-    echo "${var_name} is empty"
+prefix="NH_CONF_"
+conf_args=""
+echo "Node configuration:"
+while IFS='=' read -r -d '' var_name var_value; do
+  if [[ "$var_name" == NH_CONF* ]]; then
+    arg_name=$(get_arg_name_from_env_name "${var_name}" "${prefix}")
+    arg_value=""
+    if [ "$var_value" != "true" ]; then
+      arg_value=" ${var_value}"
+    fi
+    arg="${arg_name}${arg_value}"
+    conf_args+="${arg} "
+    echo "  ${var_name}=${var_value} -> $arg"
   fi
-done
+done < <(env -0)
 
+# Keys handling
 if [ -f "${NH_SECRET_PHRASE_PATH}" ]; then
+  path=""
+  if [ -z "${NH_CONF_BASE_PATH+}" ]; then
```

Comment:
NO sure about the end goal here, but if NH_CONF_BASE_PATH var is not set next line will produce `unbound variable` error. If it is set to "" empty string `path = --base-path`

Maybe the idea was to ` -n` vs `-z` :thinking:

Reply:
The plus sign (`+`) in this parameter expansion does the trick of avoiding unbound variable error, I tested on my machine and it's fine.

Reply:
@ysibirski I was wrong, what I wrote is not working as expected, please have a look at this fix [commit](https://github.com/HorizenLabs/NH-core/commit/3541fb1376f8a9dc5a811101190c046c4d5406a1) and let me know. Thanks